### PR TITLE
Fix unified api

### PR DESCRIPTION
**Pull request by lemaitre-aneo**

### CODE REVIEW

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs [hunk 1]
```diff
@@ -514,39 +528,51 @@ public void WaitForTasksCompletion(IEnumerable<string> taskIds)
...
+                                                  (Exception)null);
+        }
       }
     }
+    finally
+    {
+      mutex_.Release();
+    }
 
     var res = new byte[len];
```
Comment (ddubuc) [inline Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs:577]: No way to avoid a new allocation here ?

**Review by ddubuc — APPROVED** [review]
OK for me.

File: Common/ArmoniK.DevelopmentKit.Common/Disposable.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+
+namespace ArmoniK.DevelopmentKit.Common;
+
+public static class Disposable
```
Comment (aneojgurhem) [inline Common/ArmoniK.DevelopmentKit.Common/Disposable.cs:29]: Ok for me, just add the expected doc to resolve those warnings

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

**Review by ddubuc — APPROVED** [review]

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs (hunk 1 above)
Comment (lemaitre-aneo) [reply Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs]: Currently, no. Besides, it is out of scope of this PR.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**